### PR TITLE
EVG-2347 implement tag selectors for display tasks

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/project_selector.go [hunk 1]
```diff
@@ -275,6 +275,38 @@ func (t *taskSelectorEvaluator) evalSelector(s Selector) ([]string, error) {
...
+	}
+	return nil
+}
+
+func (t *displayTaskSelectorEvaluator) evalSelector(s Selector) ([]string, error) {
+	results, err := t.tagEval.evalSelector(s)
+	if err != nil {
+		return nil, errors.Wrap(err, "evaluating display task selector")
+	}
+	return results, nil
```
Comment (tychoish) [inline model/project_selector.go]: aside from the wrapping, it looks like this is just a pass through?

File: model/project_selector.go [hunk 2]
```diff
@@ -275,6 +275,38 @@ func (t *taskSelectorEvaluator) evalSelector(s Selector) ([]string, error) {
...
+}
+
+func NewDisplayTaskSelectorEvaluator(bv BuildVariant, tasks []parserTask) *displayTaskSelectorEvaluator {
+	var selectees []tagged
+	for _, t := range bv.Tasks {
+		selectees = append(selectees, &parserTask{Name: t.Name, Tags: getTags(tasks, t.Name)})
+	}
+	return &displayTaskSelectorEvaluator{
+		tagEval: newTagSelectorEvaluator(selectees),
+	}
```
Comment (tychoish) [inline model/project_selector.go]: public functions shouldn't return private types. 

having said that, I don't particularly see why there needs to be a separate type and/or why we couldn't get away with using composition (e.g. anonymous fields)

Comment (john-m-liu) [reply model/project_selector.go]: removed the separate type

File: model/project_selector.go (hunk 1 above)
Comment (john-m-liu) [reply model/project_selector.go]: yep, I removed this

**Pull request closed without merging**